### ISSUE TITLE
no-space-before-semi doesn’t check for all semicolons

### ISSUE THREAD

**Issue by lo1tuma** [eslint/eslint#1885]
There are several cases where a semicolon can occur which is not checked by `no-space-before-semi`.

Here are a few failing tests for this rule which should have an error:

```js
invalid: [ 
      { code: "while (true) { break ; }", errors: 1},
      { code: "while (true) { continue ; }", errors: 1},
      { code: "debugger ;", errors: 1},
      { code: "function foo() { return ; }", errors: 1},
      { code: "throw new Error('foo') ;", errors: 1},
      { code: "for (var i = 0 ; i < 10 ; i++) {}", errors:2 },
      // ...
]
```

Result:

```
  1) no-space-before-semi while (true) { break ; }:
     AssertionError: Should have 1 errors but had 0: []: expected 0 to equal 1

  2) no-space-before-semi while (true) { continue ; }:
     AssertionError: Should have 1 errors but had 0: []: expected 0 to equal 1

  3) no-space-before-semi debugger ;:
     AssertionError: Should have 1 errors but had 0: []: expected 0 to equal 1

  4) no-space-before-semi function foo() { return ; }:
     AssertionError: Should have 1 errors but had 0: []: expected 0 to equal 1

  5) no-space-before-semi throw new Error('foo') ;:
     AssertionError: Should have 1 errors but had 0: []: expected 0 to equal 1

  6) no-space-before-semi for (var i = 0 ; i < 10 ; i++) {}:
     AssertionError: Should have 2 errors but had 0: []: expected 0 to equal 2
```

**Comment by lo1tuma** [comment]
I also suggest to change to error message of this rule.

At the moment there are different messages related the node where the extra space before the semicolon appears:
* `"Variable declared with trailing whitespace before semicolon"`
* `"Expression called with trailing whitespace before semicolon"`

I suggest a simple error message like `"Unexpected space before semicolon"`.

**Comment by jonathanKingston** [comment]
@lo1tuma :+1: thank you
